### ISSUE TITLE
Make crossentropy optimization work with LogSoftmax

### ISSUE THREAD

**Issue by abergeron** [Theano/Theano#4596]
The crossentropy optimizations start at https://github.com/Theano/Theano/blob/master/theano/tensor/nnet/nnet.py#L1466 and they currently only look for log(softmax(...))
